### DISCUSSION TITLE
how many devices in 1-wire DS18B20 list?

### DISCUSSION THREAD

**Discussion by mtraxiot** [arendst/Tasmota#12612] [Q&A] [answered]
I would be great to include a count of devices found if people are programmatically trying to retrieve this list
eg
 **DS18B20_devices: 2,**
"DS18B20-1":{"Id":"00000566CC39","Temperature":13.3},
"DS18B20-2":{"Id":"0000059352D4","Temperature":1.2},

**Comment by sfromis** [comment, marked as answer]
Well, the "usual" JSON support in programming environments can simply parse it into an iterable data structure, without having to use indexes or name building to access the data. In fact, the driver option `DS18x20_USE_ID_AS_NAME` ditching the numbers makes more sense to me.

**Reply by mtraxiot** [reply]
As the "DS18B20" or "DS18B20-1" I assume this nane is possibly random , I'm trying to get a specific ID but the JSON structure doesn't allow iteration ie shouldn't the JSON be something like 

DS18B20: [ {"Id":"00000111CC39","Temperature":13.3},{"Id":"0000022252D4","Temperature":12.2}]

**Reply by sfromis** [reply]
Yeah, there is no reliable pairing from a specific sensor to the number, that's why I think that it makes more sense to entirely ditch the number-part of the sensor name (replacing by part of the Id).

If you want to find a specific one of the sensors, you could build a binary with `#define DS18x20_USE_ID_AS_NAME` to have the Id (part of it) replace the number in the names.

What I've used to programmatically step through parsed JSON data makes it easy to iterate the data structure without it being an array. No indexes was meant literally, just iterating through the available sensors.

While I do see the logic of having an array of DS18x20 sensors, this would be a major change in the data format, screwing badly with existing automations parsing the data.

**Reply by mtraxiot** [reply]
I understand the impact which is why just having the  "DS18B20_devices:  _**count**_" would be only a simple low impact change and at least you can determine what to parse.

**Reply by barbudor** [reply]
Nothing is low impact in an memory constrained environment where many features are packed.
Considering the length of MQTT message is limited in Tasmota to save RAM, surcharging the JSON with a (sorry to call it like that) _useless_ field would also reduce the number of DS18x20 that could be supported.
You programming language should have a `len()` or `length()` method or something similar on the array/list. Python and javascript have.
And as sfromis said, use object iteration where no index is used. This is better programming.
